binary_tree.c

```diff
@@ -0,0 +1,410 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include "node.h"
+#include "bfs_queue.h"
+
+typedef struct {
+    struct Node* root;
+} Tree;
+
+
+Tree tree_create(); // returns empty binary tree
+// void tree_free(Tree* self); // deallocates tree
+
+struct Node* tree_create_node(int);
+void tree_delete_node(Tree*, struct Node*);
+
+int tree_get_node_val(struct Node*);
```

Comment:
Do you really need these functions?

Reply:
I copied it from the very basic tree interface.
See no problems if it remains here.
I tested it as other functions
Wdyt?

Reply:
Well, instead of calling this function we can always do `node->data`, and it makes no difference.